bundles/config/org.eclipse.smarthome.config.xml/src/main/java/org/eclipse/smarthome/config/xml/osgi/XmlDocumentBundleTracker.java

```diff
@@ -34,27 +48,26 @@
  * @author Michael Grammling - Initial Contribution
  * @author Benedikt Niehues - Changed resource handling so that resources can be
  *         patched by fragments.
+ * @author Simon Kaufmann - Tracking of remaining bundles
  *
  * @param <T>
  *            the result type of the conversion
  */
+@SuppressWarnings({ "unchecked", "rawtypes" })
```

Comment:
> Using this changes we don't need to supress warnings

Right. That was my initial thought too. 
And then I realized that it won't run on OSGi 4.2 anymore (which doesn't have generics in its signatures) and I end up having some very-not-so-amused colleagues 😉 
But I agree, it sucks!

> Do we really suppress all that warnings for the whole class?

You are right, it's safer to do that on variable level. Will change that.